### DISCUSSION TITLE
V8 BETA

### DISCUSSION THREAD

**Discussion by marcus-j-davies** [zwave-js/node-red-contrib-zwave-js#248] [Q&A]
@fabnavigator

V8 BETA 1 is out!

**Note!**
This is using V10 of ZWave-JS (which is also in BETA), when V10 of ZWave-JS steps out of BETA into stable (for me it already is)
I will of course follow suite shortly after - providing no major issues with my V8 BETAs.

Pre warning: **BACK UP YOUR SYSTEM!**

Guide - adjust where necessary
```
cd ~/.node-red
npm install node-red-contrib-zwave-js@beta --force
sudo systemctl restart node-red 
```

@lurendrejer 
I am going to tag you in this also - not sure you do a lot of testing, but just in case 😅 

Both,
Please see the change log before installing the beta.
https://github.com/zwave-js/node-red-contrib-zwave-js/blob/8.0.0/CHANGELOG.md

**Comment by lurendrejer** [comment]
Will be checking this out when I get home in a few days time

**Reply by fabnavigator** [reply]
I think I found an issue but would like to revert back to the previous beta version to verify. What is the best way to do that?

**Reply by marcus-j-davies** [reply]
Navigate to the /.node-red dir 

Then install any version you want.

npm install node-red-contrib-zwave-js@7.2.1-beta.6 restart node red after. You can double click the controller node to verify the version that is installed

**Reply by fabnavigator** [reply]
That didn't work.
```
fabnav@docker-hass:~/.node-red $ npm install node-red-contrib-zwave-js@7.2.1-beta.5
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/get-entity-id/-/get-entity-id-1.0.12.tgz - Not found
npm ERR! 404
npm ERR! 404  'get-entity-id@https://registry.npmjs.org/get-entity-id/-/get-entity-id-1.0.12.tgz' is not in this registry.
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/fabnav/.npm/_logs/2022-08-08T20_27_34_411Z-debug-0.log
```

**Reply by marcus-j-davies** [reply]
Might have been a temporary glitch with the NPM registry - Just tried it with no prob.

try using `--force`

```
md@Marcuss-Mac-mini untitled folder % npm install node-red-contrib-zwave-js@7.2.1-beta.6 --force
npm WARN using --force I sure hope you know what you are doing.

> @serialport/bindings-cpp@10.7.0 install /Users/md/Documents/untitled folder/node_modules/@serialport/bindings-cpp
> node-gyp-build

npm WARN saveError ENOENT: no such file or directory, open '/Users/md/Documents/untitled folder/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/Users/md/Documents/untitled folder/package.json'
npm WARN untitled folder No description
npm WARN untitled folder No repository field.
npm WARN untitled folder No README data
npm WARN untitled folder No license field.

+ node-red-contrib-zwave-js@7.2.1-beta.6
added 141 packages from 207 contributors and audited 141 packages in 17.575s

28 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

md@Marcuss-Mac-mini untitled folder % 

```

**Reply by fabnavigator** [reply]
That didn't help.

**Reply by fabnavigator** [reply]
In any case, I see that the ```normalizedObject.label``` for my thermostat changed from ```Cooling``` to ```Setpoint (Cooling)``` in 8.0 beta.
I also saw changes to the normilazed values for my door sensor.

**Reply by marcus-j-davies** [reply]
No idea why `get-entity-id` is trying to get installed, as it was pulled from npm -  Its not erroring for me. maybe there is something in your `package.json` (same folder) that is indirectly referencing it.

try this

```
wget https://registry.npmjs.org/node-red-contrib-zwave-js/-/node-red-contrib-zwave-js-7.2.1-beta.6.tgz
npm install ./node-red-contrib-zwave-js-7.2.1-beta.6.tgz  
```

**Reply by marcus-j-davies** [reply]
> I also saw changes to the normilazed values for my door sensor.

Yep some of the labels were updated in V10 of the driver, the labels come directly from the driver.

**Reply by fabnavigator** [reply]
```Yep some of the labels were updated in V10 of the driver, the labels come directly from the driver.``` okay.
Installing older version, still NG.

**Reply by fabnavigator** [reply]
It must be late for you. This isn't a rush.

**Reply by marcus-j-davies** [reply]
what is in your `package.json`

**Reply by marcus-j-davies** [reply]
`get-entity-id` was pulled from NPM a while back

https://registry.npmjs.org/get-entity-id

**Reply by fabnavigator** [reply]
```
{
    "name": "node-red-project",
    "description": "A Node-RED Project",
    "version": "0.0.1",
    "private": true,
    "dependencies": {
        "bufferutil": "^4.0.6",
        "get-entity-id": "^1.0.12",
        "node-red-contrib-eztimer": "^1.2.7",
        "node-red-contrib-fs": "^1.4.1",
        "node-red-contrib-postgres-variable": "^0.1.5",
        "node-red-contrib-zwave-js": "^8.0.0-beta.1",
        "node-red-dashboard": "~3.1.7",
        "node-red-node-email": "~1.15.1"
    }
}```

**Reply by fabnavigator** [reply]
Should I just remove it?

**Reply by marcus-j-davies** [reply]
I would.... - its doesn't exist

**Reply by marcus-j-davies** [reply]
once removed, you should be able to redo

`npm install node-red-contrib-zwave-js@7.2.1-beta.6 --force`

NPM will install anything that is not installed, and referenced in `package.json`

**Reply by fabnavigator** [reply]
Yep. Now it works. But with you telling me about the name changes, I no longer need the older version, but it's nice to know that I can get back when I want to. I see that even on your last version, I'm still getting the V10 driver version. What if I wanted to walk that back?

**Reply by marcus-j-davies** [reply]
Doh! V10 Beta probably satisfies semver of **^9.6.0** of the driver - semver doesn't handle -beta tags all that well IMO

You can clear it all out, if you wanted to get back to the last (pre V8) Beta.

Follow it to the letter..

 - Stop Node RED
 - Navigate to NR's home directory **~/.node-red** (if you have not changed it)
 - Delete **package-lock.json**
 - Open **package.json**
 - Remove any package/line in this file that are for **zwave-js**, **serialport** and **node-red-contrib-zwave-js**
 - Navigate to **~/.node-red/node_modules**
 - In this folder delete any folder for **zwave-js**, **@ zwave-js,** **@ AlCalzone,** **serialport**, **@ serialport** and  **node-red-contrib-zwave-js**
 - Navigate back to **~/.node-red**
 - run `npm install `(this will recreate the lock)
 - `run npm install node-red-contrib-zwave-js@7.2.1-beta.6`
 - start up Node RED

**Reply by marcus-j-davies** [reply]
V8 is when I switched to V10 of the driver (7.2.1, moved to V8 once I learnt of the breaking changes in v10 - and wasn't worth realising 7.2.1 because of it)

**Reply by fabnavigator** [reply]
@marcus-j-davies,

What about the ```alcalzone-shared``` directory?

**Reply by marcus-j-davies** [reply]
Yup - that can go also 👍

**Reply by marcus-j-davies** [reply]
as long as you retain **zwave-js-cache** you have nothing to worry about.

**Reply by marcus-j-davies** [reply]
**zwave-js-cache**  = All your Zwave Network data

**Comment by marcus-j-davies** [comment]
@fabnavigator 
BETA 2 is up! (uses BETA 7 of V10 of the driver) - that should address the initial issues 

`npm install node-red-contrib-zwave-js@beta --force`

**Reply by fabnavigator** [reply]
@marcus-j-davies, Looks good so far.

**Comment by lurendrejer** [comment]
Finally got around to installing the beta.
Everything has seemed stable for the last 24 hours and associations-UI is working again.

Thank you very much!

**Reply by marcus-j-davies** [reply]
Excellent @lurendrejer 

I will be aiming for a stable release soon.

Still one more thing todo, then will push v8 as stable 👍

**Comment by marcus-j-davies** [comment]
@fabnavigator @lurendrejer 

BETA 5 is now available.
I am hoping this is it before taking it to a stable release.

within your `.node-red` directory
`npm install node-red-contrib-zwave-js@beta --force`
Restart Node RED

if you haven't already, please see change log.
https://github.com/zwave-js/node-red-contrib-zwave-js/blob/8.0.0/CHANGELOG.md

Would be much appreciated for you to ensure the final BETA is trouble free.

**Reply by fabnavigator** [reply]
@marcus-j-davies I'm in. So far so good.

**Reply by marcus-j-davies** [reply]
BETA 5 should see a huge performance increase in the UI when loading node info

**Reply by fabnavigator** [reply]
@marcus-j-davies It doesn't take long, but I don't recall it taking that long before.

**Reply by marcus-j-davies** [reply]
Scene Actuator devices (for example) caused 300+ calls to the internal web services when selected, this at times took a few seconds to complete, if during that time - you were navigating the UI - there were delays (stopping a FW update for example was delayed)

Every device you now select only 1 call is made to the web service (irrelevant of how many properties it needs to gain info for)

you can see it happening - Open up the UI monitor, and select a node - you will only witness 1 call, prior to BETA 5, this was at least 40


This results in a much quicker UI

**Reply by marcus-j-davies** [reply]
This final BETA also takes advantage of the new function `getLastEvents` - to show last seen in the UI.
![image](https://user-images.githubusercontent.com/55892693/190911161-ca19a03b-10bc-41a9-b754-295346e80311.png)

https://github.com/zwave-js/node-red-contrib-zwave-js/wiki/Driver-API#getlastevents

**Reply by fabnavigator** [reply]
@marcus-j-davies I like that new feature, though you have to refresh the node list to see the latest information.

**Reply by marcus-j-davies** [reply]
I don't foresee the UI being left open long enough to have much of a difference 😅 
clicking the Network ID also reloads it all

**Reply by marcus-j-davies** [reply]
@fabnavigator 

any issues to report? - I'm preparing for stable release if not.

**Reply by fabnavigator** [reply]
@marcus-j-davies Well, I can't say this is due to the release, but I see that my daily job to run through the nodes and do a health check isn't getting through all of the nodes recently. I kicked it off manually and it got this error:
```
ZWaveError: The node failed to decode the message. (ZW1405)
    at Driver.sendMessage (/home/fabnav/.node-red/node_modules/zwave-js/src/lib/driver/Driver.ts:3963:23)
    at Driver.sendCommandInternal (/home/fabnav/.node-red/node_modules/zwave-js/src/lib/driver/Driver.ts:4154:28)
    at Driver.sendSupervisedCommand (/home/fabnav/.node-red/node_modules/zwave-js/src/lib/driver/Driver.ts:4199:27)
    at Driver.sendCommand (/home/fabnav/.node-red/node_modules/zwave-js/src/lib/driver/Driver.ts:4251:30)
    at Proxy.startNodeTest (/home/fabnav/.node-red/node_modules/@zwave-js/cc/src/cc/PowerlevelCC.ts:133:24)
    at ZWaveNode.testPowerlevel (/home/fabnav/.node-red/node_modules/zwave-js/src/lib/node/Node.ts:4266:13)
    at executor (/home/fabnav/.node-red/node_modules/zwave-js/src/lib/node/Node.ts:4478:32)
    at discreteLinearSearch (/home/fabnav/.node-red/node_modules/@zwave-js/shared/src/utils.ts:193:24)
    at ZWaveNode.checkLifelineHealth (/home/fabnav/.node-red/node_modules/zwave-js/src/...
```
Does that mean something to you?

**Reply by marcus-j-davies** [reply]
That's just an S2 clash - its can occur if too many things are happening.
try and space your health checks out (delay for like 1 min after the last result) - and do not run > 1 at any given time.

**Reply by marcus-j-davies** [reply]
Health Checks are network intensive beasts

**Reply by fabnavigator** [reply]
I'm doing three rounds per node. I don't think I can add a delay between the rounds.

**Reply by marcus-j-davies** [reply]
after you get the result, add a delay before asking for the next set of rounds

**Reply by fabnavigator** [reply]
I will try that. My sniffer picked up a ```Routed Error``` soon before everything errored out.

**Reply by marcus-j-davies** [reply]
Nice use of the sniffer 😉  - it all points to congestion, in normal use it goes unnoticed.
as normal CC commands are retried.

**Reply by marcus-j-davies** [reply]
I will be releasing very  soon

**Reply by fabnavigator** [reply]
I haven't seen any other issues. I'll let you know how the health check w/ delay worked out. In like an hour...

**Reply by fabnavigator** [reply]
It failed with the one minute delay. I'll switch to a single round and see how that does.

**Reply by fabnavigator** [reply]
Even with a single round and the one minute delay between nodes I can't get through my 27 nodes.  I am starting to think that something is wrong. I havn't made any changes to my nodes recently, and no increased traffic. I wonder if I should roll back to a previous version and see if that makes a difference.

**Reply by marcus-j-davies** [reply]
Ok....

I think I know what it is.
just ran a health check on a FLiRS device.

![Screenshot 2022-09-25 at 16 56 27](https://user-images.githubusercontent.com/55892693/192152943-0d6f9ec2-99f1-41c7-8da8-99a7ad3d448d.png)

Are you running these HC's on devices that are awake (sun)  -but have a battery level like this?


![Screenshot 2022-09-25 at 16 57 44](https://user-images.githubusercontent.com/55892693/192152978-610a1511-c6d0-4ed6-a638-e187650bb42c.png)

**Reply by fabnavigator** [reply]
No. I don't run the health check on battery powered devices.

It always fails on an S2 node, but different ones. No issues on the unsecured nodes. Node 105 below was the node that failed this last time.

![image](https://user-images.githubusercontent.com/16625904/192154988-9ac7d03b-2151-46b6-9d87-203e0cff6fad.png)

**Reply by marcus-j-davies** [reply]
So,
any S2 node (and always)?

**Reply by fabnavigator** [reply]
No. The first two S2 nodes haven't had an issue. The third has failed multiple times, but this last time the fourth S2 node failed.

**Reply by fabnavigator** [reply]
Everything was running w/o errors, three rounds, no delay between nodes for almost a month, until 9/19. Tnen node 105 failed, and failed again on 9/21. 9/20 and 9/22 everything passed. On 9/23 node 42 (first S2) failed.

**Reply by marcus-j-davies** [reply]
Ok,

Can I ask you raise a bug report on the driver repo - remember to include a log file of the error happening so we can see events  leading up to it 😅 

I have an S2 FLiRS device - that never works with health checks (it did previously), so can add some further logs to it if needed.
It's probably some recent changes to the driver 🤷

**Comment by marcus-j-davies** [comment]
V8 released. 

Should appear in Node RED shortly!

**Comment by marcus-j-davies** [comment]
I'm gonna look this thread  - given we are now out of BETA 😄
